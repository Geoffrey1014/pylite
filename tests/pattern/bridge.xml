lambda
  name __init__
  param_count 0
  local_count 1
  label_count 3
  insn_vector
    new_class c0
    store_global c0
    load_const c1
    load_global c0
    store_field c2
    new_class c3
    store_global c3
    load_const c4
    load_global c3
    store_field c2
    new_class c5
    store_global c5
    load_const c6
    load_global c5
    store_field c7
    load_const c8
    load_global c5
    store_field c9
    new_list 2
    load_global c0
    call 0
    load_const c10
    load_const c11
    load_const c12
    load_global c5
    call 4
    load_const c13
    dup_top 2
    store_item
    load_global c3
    call 0
    load_const c14
    load_const c15
    load_const c16
    load_global c5
    call 4
    load_const c12
    dup_top 2
    store_item
    store_global c17
    load_global c17
    iterator
    store_local -1
    jmp L1
    label L0
    store_global c18
    load_global c18
    load_field c9
    call 0
    pop
    label L1
    load_local -1
    next L0
    pop
    label L2
    load_const c19
    return
  const_vector
    DrawingAPI1
    lambda
      name draw_circle
      param_count 4
      local_count 0
      label_count 0
      insn_vector
        load_local 3
        load_local 2
        load_local 1
        new_tuple 3
        load_const c0
        mod
        print
        load_const c1
        return
      const_vector
        "API1.circle(%d, %d, %d)"
        None
      guard_vector
      line_number_map 4 4 4 4 4 4 4 3 3
    draw_circle
    DrawingAPI2
    lambda
      name draw_circle
      param_count 4
      local_count 0
      label_count 0
      insn_vector
        load_local 3
        load_local 2
        load_local 1
        new_tuple 3
        load_const c0
        mod
        print
        load_const c1
        return
      const_vector
        "API2.circle(%d, %d, %d)"
        None
      guard_vector
      line_number_map 9 9 9 9 9 9 9 8 8
    Circle
    lambda
      name __init__
      param_count 5
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        store_field c0
        load_local 2
        load_local 0
        store_field c1
        load_local 3
        load_local 0
        store_field c2
        load_local 4
        load_local 0
        store_field c3
        load_const c4
        return
      const_vector
        x
        y
        radius
        drawing_api
        None
      guard_vector
      line_number_map 14 14 14 15 15 15 16 16 16 17 17 17 13 13
    __init__
    lambda
      name draw
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        load_local 0
        load_field c1
        load_local 0
        load_field c2
        load_local 0
        load_field c3
        load_field c4
        call 3
        pop
        load_const c5
        return
      const_vector
        radius
        y
        x
        drawing_api
        draw_circle
        None
      guard_vector
      line_number_map 21 21 21 21 21 21 21 21 21 21 21 20 20
    draw
    3.000000
    2.000000
    1.000000
    0.000000
    6.000000
    5.000000
    4.000000
    shapes
    shape
    None
  guard_vector
  line_number_map 2 2 3 3 3 7 7 8 8 8 12 12 13 13 13 20 20 20 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 23 24 24 24 24 24 24 25 25 25 25 24 24 24 24 24 2 2
