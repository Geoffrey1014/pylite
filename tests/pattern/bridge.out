API1.circle(1, 2, 3)
API2.circle(4, 5, 6)
